### DISCUSSION TITLE
Verifying Contract deployed through script

### DISCUSSION THREAD

**Discussion by Ibby-debug** [foundry-rs/foundry#6130] [Help]
I'm running a simple script to deploy my contract. 

```
forge script script/Deploy.s.sol  --broadcast --rpc-url $AVALANCHE_DEVNET_CONNECTION --private-key $DEPLOYER --etherscan-api-key $SNOWTRACE_API_KEY --verify 
```

The contract gets deployed successfully.  However, I get this error back! 
```error 
We haven't found any matching bytecode for the following contracts: [0x....].

This may occur when resuming a verification, but the underlying source code or compiler version has changed.
```

**Comment by ratankaliani** [comment]
One of my libraries imported another library using a relative import (e.g. `@openzeppelin-contracts`). Because of some versioning issues (where I needed to install `openzeppelin-contracts` at a specific version in my repository's `gitsubmodules`), I decided to set the relative import path in my `remappings.txt` as follows, which caused issues when attempting to verify a proxy contract imported from `openzeppelin`.

```
@sp1-contracts/=lib/sp1-contracts/contracts/src/
@succinctx/=lib/succinctx/contracts/src
@openzeppelin/=lib/succinctx/contracts/lib/openzeppelin-contracts/
@openzeppelin-upgradeable/=lib/succinctx/contracts/lib/openzeppelin-contracts-upgradeable/
```

After trying a bunch of different things recommended by https://github.com/foundry-rs/foundry/issues/5170, https://github.com/foundry-rs/foundry/issues/4865, I decided to install the dependencies that I needed directly. I retrieved the version required by my `succinctx` dependency for the openzeppelin contracts and installed it in my Foundry project, rather than using `remappings.txt` to access a nested submodule.

After changing it to this, my Proxy contract from `openzeppelin` verified:

```
@sp1-contracts/=lib/sp1-contracts/contracts/src/
@succinctx/=lib/succinctx/contracts/src
@openzeppelin/=lib/openzeppelin-contracts/
@openzeppelin-upgradeable/=lib/openzeppelin-contracts-upgradeable/
```

**Reply by Ibby-debug** [reply]
Banger. Yeah I remember still getting a tough time verifying contracts while deploying, will try this out and let you know! 

Also curious to know. Are you still hiring for the Frontend Developer role ? I'm really passionate about crypto and I'm very good with working across the full stack. (Frontend ----> Bakcend ----> Contract).
